gmime/part.go

```diff
@@ -103,6 +103,14 @@ func (p *Part) Headers() textproto.MIMEHeader {
 	return nil
 }
 
+// GetContentID returns the content ID (if there is one) per mime part
+// we expect all attachments to have a content-id
+func (p *Part) GetContentID() string {
+	cCID := C.g_mime_object_get_content_id(p.gmimePart)
```

Comment:
you might need to something similar to filename function
`if p.gmimePart == nil {
		return ""
	}
	if !gobool(C.gmime_is_part(p.gmimePart)) {
		return ""
	}`

Reply:
Great call, and thank you for the code sample.

I'll get this updated.